### PR TITLE
Feature: New gitlab webhook

### PR DESCRIPTION
**Pull request by alexrodfe**
## Description

- Added new predefined process gitlab webhook
- Updated docs
- Fixed an issue with tests, they were not correctly checking the type of the response being created
- Fixed cronjob tests

## Motivation and Context

Users will now be able to use a Gitlab webhook if they want to trigger any workflow from a Gitlab repo event.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other changes (ci configuration, documentation or any other kind of changes)

## Checklist:

- [X] I have created tests for my code changes, and the tests are passing.
- [ ] I have executed the pre-commit hooks locally.
- [ ] My change requires a change to the documentation (create a new issue if the documentation has not been updated).
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Review by kafkaphoenix — APPROVED** [review]
good job!

File: github-webhook-trigger/README.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+| webhook_events | no        | str  | Possible options (comma separated): push, pull, release, workflow_dispatch, workflow_run      |
+| github_secret  | yes       | str  | Github's webhook secret
+
+## Output (JSON)
+
+| Key       | Type | Value                                                                  |
+|-----------|------|------------------------------------------------------------------------|
+| requestID | str  | A string uuid                                                          |
+| eventUrl  | str  | The url from the repo triggering the event                             |
+| event     | str  | The name of the event that has occurred                                |
```
Comment (dfinteligenz) [inline github-webhook-trigger/README.md:32]: I'd add a real example of the output below the table,  in JSON format.

File: gitlab-webhook-trigger/README.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+| webhook_events | no        | str  | Possible options (comma separated): push, merge_request, comment, tag   |
+| gitlab_secret  | yes       | str  | Gitlab's webhook secret
+
+## Output (JSON)
+
+| Key       | Type | Value                                                                  |
+|-----------|------|------------------------------------------------------------------------|
+| requestID | str  | A string uuid                                                          |
+| eventUrl  | str  | The url from the repo triggering the event                             |
+| event     | str  | The name of the event that has occurred                                |
```
Comment (dfinteligenz) [inline gitlab-webhook-trigger/README.md:31]: I'd add a real example of the output below the table, in JSON format.

File: gitlab-webhook-trigger/app.yaml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+nats:
+  url: 'nats://localhost:4222'
+  stream: github-trigger
+  output: github-trigger-output
+  object_store: object-store
+
+centralized_configuration:
+  product:
+    bucket: product
```
Comment (dfinteligenz) [inline gitlab-webhook-trigger/app.yaml:16]:
```suggestion
centralized_configuration:
  global:
    bucket: global
  product:
    bucket: product
```

File: gitlab-webhook-trigger/minikube_build.sh [hunk 4]
```diff
@@ -0,0 +1,4 @@
+#!/bin/sh
+
+CGO_ENABLED=0 go build -o build/process main.go
+minikube image build -t kai-processes/github-webhook-trigger . -f develop.Dockerfile -p kai-local
```
Comment (dfinteligenz) [inline gitlab-webhook-trigger/minikube_build.sh:4]:
```suggestion
minikube image build -t kai-processes/gitlab-webhook-trigger . -f develop.Dockerfile -p kai-local
```

File: gitlab-webhook-trigger/sonar-project.properties [hunk 5]
```diff
@@ -0,0 +1,17 @@
+sonar.projectKey=konstellation-io_kai-processes_github_webhook_trigger
+sonar.organization=konstellation-io
+
+# This is the name and version displayed in the SonarCloud UI.
+sonar.projectName=Kai Processes - Github Webhook Trigger
```
Comment (dfinteligenz) [inline gitlab-webhook-trigger/sonar-project.properties:5]:
```suggestion
sonar.projectName=Kai Processes - GitLab Webhook Trigger
```

File: gitlab-webhook-trigger/sonar-project.properties [hunk 6]
```diff
@@ -0,0 +1,17 @@
+sonar.projectKey=konstellation-io_kai-processes_github_webhook_trigger
```
Comment (dfinteligenz) [inline gitlab-webhook-trigger/sonar-project.properties:1]:
```suggestion
sonar.projectKey=konstellation-io_kai-processes_gitlab_webhook_trigger
```

**Review by dfinteligenz — CHANGES_REQUESTED** [review]

File: github-webhook-trigger/README.md [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+One being the events the webhook will listen to (_webhook_events_), the other the github secret needed to interact with the github repo (_github_secret_).
+
+! Github repository needs to be configured also to expose events to "/webhooks" please check [webhook_guide](https://docs.github.com/webhooks/) for more information.
+
+### Config
+
+| Key            | Optional  | Type | Value                                                                                         |
+|----------------|-----------|------|-----------------------------------------------------------------------------------------------|
+| webhook_events | no        | str  | Possible options (comma separated): push, pull, release, workflow_dispatch, workflow_run      |
+| github_secret  | yes       | str  | Github's webhook secret
```
Comment (dfinteligenz) [inline github-webhook-trigger/README.md:24]:
```suggestion
| Key            | Optional  | Type | Value                                                                                         |
|----------------|-----------|------|-----------------------------------------------------------------------------------------------|
| webhook_events | no        | str  | Possible options (comma separated): push, pull, release, workflow_dispatch, workflow_run      |
| github_secret  | yes       | str  | Github's webhook secret.  |
```

**Review by dfinteligenz — APPROVED** [review]

**Pull request merged**